californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -248,7 +235,7 @@ public boolean registerOutboundRequest(final Exchange exchange) {
 			throw new IllegalArgumentException("exchange does not contain a request");
 		} else {
 			int mid = registerWithMessageId(exchange, exchange.getCurrentRequest());
-			if (mid > Message.NONE) {
+			if (Message.NONE != mid) {
```

Comment:
just out of interest: do you suspect a problem with the original comparison? FMPOV both alternatives should do the trick, wouldn't they?